clang/lib/CIR/Dialect/Transforms/TargetLowering/TargetInfo.h

```diff
@@ -19,14 +19,16 @@
 #include "clang/CIR/Target/AArch64.h"
 #include "clang/CIR/Target/x86.h"
 
+using namespace ::cir;
+
 namespace mlir {
 namespace cir {
```

Comment:
@bcardosolopes what's the current namespace scheme? What should be in `::mlir::cir` vs `::cir`?

Reply:
If it's dialect related I usually put thing in mlir::cir, and ::cir is for the rest (perhaps at some point we should migrate to clang::codegen::cir or something, but not there yet)

Reply:
Also, please don't use `using namespace` inside headers.

Reply:
> I usually put thing in mlir::cir, and ::cir is for the rest

I don't think `::mlir::cir` should exist at all. AFAIK, a declaration is under the `mlir` namespace when declared within the MLIR project. We should probably follow FIR's style and skip the `::mlir` part, declaring everything within `::cir::<namespace>`. This might even end up as a requirement for upstreaming CIR. :thinking: